src/main/java/com/minecolonies/coremod/colony/buildings/modules/settings/BoolSetting.java

```diff
@@ -0,0 +1,113 @@
+package com.minecolonies.coremod.colony.buildings.modules.settings;
+
+import com.ldtteam.blockout.Color;
+import com.ldtteam.blockout.controls.ButtonImage;
+import com.ldtteam.blockout.controls.Text;
+import com.ldtteam.blockout.views.Box;
+import com.minecolonies.api.colony.buildings.modules.settings.ISetting;
+import com.minecolonies.api.colony.buildings.modules.settings.ISettingKey;
+import com.minecolonies.api.colony.buildings.modules.settings.ISettingsModuleView;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+
+import static com.minecolonies.api.util.constant.WindowConstants.OFF;
+import static com.minecolonies.api.util.constant.WindowConstants.ON;
+
+/**
+ * Stores a boolean setting.
+ */
+public class BoolSetting implements ISetting
+{
+    /**
+     * The value of the setting.
+     */
+    private boolean value;
+
+    /**
+     * Default value of the setting.
+     */
+    private boolean defaultValue;
+
+    /**
+     * Create a new boolean setting.
+     * @param init the initial value.
+     */
+    public BoolSetting(final boolean init)
+    {
+        this.value = init;
+        this.defaultValue = init;
+    }
+
+    /**
+     * Create a new boolean setting.
+     * @param value the value.
+     * @param def the default value.
+     */
+    public BoolSetting(final boolean value, final boolean def)
+    {
+        this.value = value;
+        this.defaultValue = def;
+    }
+
+    /**
+     * Get the setting value.
+     * @return the set value.
+     */
+    public boolean getValue()
+    {
+        return value;
+    }
+
+    /**
+     * Get the default value.
+     * @return the default value.
+     */
+    public boolean getDefault()
+    {
+        return defaultValue;
+    }
+
+    @OnlyIn(Dist.CLIENT)
+    @Override
+    public void addHandlersToBox(final ISettingKey<?> key, final Box box, final ISettingsModuleView settingsModuleView)
+    {
+        final ButtonImage image;
+        if (box.findPaneByID("trigger") == null)
```

Comment:
we really should stop creating bigger guis/elements from code and use layouts instead
(nothing you need to change, just generic opinion)